hfile/reader.go

```diff
@@ -254,3 +265,74 @@ func (r *Reader) GetIterator() *Iterator {
 		return NewIterator(r)
 	}
 }
+
+// TODO(benjy): Write tests for this, once the writer supports writing the FileInfo block.
+func (r *Reader) readFileInfo(data []byte) (err error) {
+	type outOfBounds struct {}
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch r.(type) {
+			case outOfBounds:
+				err = errors.New("Tried to read past end of FileInfo block!")
+			default:
+				panic(r)
+			}
+		}
+	}()
+
+	r.FileInfo.InfoFields = make(map[string]string)
+
+	fileInfoBlockEnd := r.DataIndexOffset
+
+	p := r.FileInfoOffset
+
+	if p == fileInfoBlockEnd {
+		log.Println("[Reader.readFileInfo] No FileInfo block found. Skipping.")
+		return nil
+	}
+
+	// Safely move the read pointer.
+	advance := func(increment int) {
```

Comment:
I made it local so that it can access p from its closure, instead of having to pass it in. I'll look at bytes.Reader.